File: dell-training/weekly_meetings/sep1st2022/sdk-app-starter/java/templates/Monitoring.java
package com.opsramp.gateway.app.actions.impl.monitor;

import java.util.List;

import com.opsramp.app.content.core.actions.Monitoring;
import com.opsramp.app.content.core.handlers.context.RequestContext;
import com.opsramp.app.content.core.payload.MetricPayload;
import com.opsramp.gateway.app.domain.TargetMonitoring;

public class {{nativeType}}Monitor implements Monitoring {

	public void monitor(RequestContext requestContext) throws Exception {
		List<MetricPayload> metricPayloads = new TargetMonitoring().process{{nativeType}}Metrics(requestContext);
		MonitoringUtil.publishMetricsandAlerts(requestContext, metricPayloads);
	}
}

File: dell-training/weekly_meetings/sep1st2022/sdk-app-starter/python/templates/monitoring.py
from core import AbstractMonitoring, GatewayToCloudMessage
from target import TargetMonitoring


import logging
logger = logging.getLogger(__name__)


class {{nativeType}}Monitoring(AbstractMonitoring):

    def monitor(self, requestcontext):
        try:
            logger.debug("Collecting {{nativeType}} metrics")
            {{nativeType}}_metrics = TargetMonitoring().process_{{nativeType.lower()}}_metrics(requestcontext)
            if {{nativeType}}_metrics is not None and len({{nativeType}}_metrics) > 0:
                GatewayToCloudMessage.publish_metrics(requestcontext.context["http_headers"],
                                                      {{nativeType}}_metrics)
            else:
                logger.info("metrics json not found")
                logger.debug("{{nativeType}} monitoring json " + str({{nativeType}}_metrics))
        except Exception as e:
            raise Exception(str(e))

